cvat-ui/src/components/import-dataset/import-dataset-modal.tsx

```diff
@@ -66,67 +277,71 @@ function ImportDatasetModal(props: StateToProps): JSX.Element {
+        instanceType: '',
+        file: null,
+        selectedLoader: null,
         useDefaultSettings: true,
-    } as UploadParams);
-    const [resource, setResource] = useState('');
+        defaultStorageLocation: StorageLocation.LOCAL,
+        defaultStorageCloudId: undefined,
+        helpMessage: '',
+        selectedSourceStorageLocation: StorageLocation.LOCAL,
+        uploadParams: {
+            convMaskToPoly: true,
+            useDefaultSettings: true,
+        } as UploadParams,
+        resource: '',
+    });
+
+    const {
+        instanceType,
+        file,
+        selectedLoader,
+        useDefaultSettings,
+        defaultStorageLocation,
+        defaultStorageCloudId,
+        helpMessage,
+        selectedSourceStorageLocation,
+        uploadParams,
+        resource,
+    } = state;
 
     useEffect(() => {
         if (instanceT === 'project') {
-            setResource('dataset');
+            dispatchReducer(reducerActions.setResource('dataset'));
         } else if (instanceT === 'task' || instanceT === 'job') {
-            setResource('annotation');
+            dispatchReducer(reducerActions.setResource('annotation'));
         }
     }, [instanceT]);
 
     const isDataset = useCallback((): boolean => resource === 'dataset', [resource]);
     const isAnnotation = useCallback((): boolean => resource === 'annotation', [resource]);
 
-    useEffect(() => {
-        setUploadParams({
-            ...uploadParams,
-            resource,
-            sourceStorage: {
-                location: defaultStorageLocation,
-                cloudStorageId: defaultStorageCloudId,
-            } as Storage,
-        } as UploadParams);
-    }, [resource, defaultStorageLocation, defaultStorageCloudId]);
-
     const isProject = useCallback((): boolean => instance instanceof core.classes.Project, [instance]);
     const isTask = useCallback((): boolean => instance instanceof core.classes.Task, [instance]);
 
     useEffect(() => {
         if (instance) {
-            setDefaultStorageLocation(instance.sourceStorage.location);
-            setDefaultStorageCloudId(instance.sourceStorage.cloudStorageId);
+            dispatchReducer(reducerActions.setDefaultStorageLocation(instance.sourceStorage.location));
+            dispatchReducer(reducerActions.setDefaultStorageCloudId(instance.sourceStorage.cloudStorageId));
             let type: 'project' | 'task' | 'job' = 'job';
 
             if (isProject()) {
                 type = 'project';
             } else if (isTask()) {
                 type = 'task';
             }
-            setInstanceType(`${type} #${instance.id}`);
+            dispatchReducer(reducerActions.setInstanceType(`${type} #${instance.id}`));
         }
     }, [instance, resource]);
 
     useEffect(() => {
-        setHelpMessage(
+        dispatchReducer(reducerActions.setHelpMessage(
             // eslint-disable-next-line prefer-template
```

Comment:
I think we can remove this `eslint disable` since the template literals are already used  in the string below